tests/integration/test_non_user_endpoints.py

```diff
@@ -175,6 +175,18 @@ def test_artist_search_with_market(self):
         self.assertTrue(len(results['artists']['items']) > 0)
         self.assertTrue(results['artists']['items'][0]['name'] == 'Weezer')
 
+    def test_artist_search_with_markets(self):
+        results = self.spotify.search(q='weezer', type='artist', market=['GB', 'US', 'AU'])
+        self.assertTrue('artists' in results)
+        self.assertTrue(len(results['artists']['items']) > 0)
+        self.assertTrue(results['artists']['items'][0]['name'] == 'Weezer')
+
+    def test_artist_search_with_all_markets(self):
+        results = self.spotify.search(q='weezer', type='artist', market='ALL')
+        self.assertTrue('artists' in results)
+        self.assertTrue(len(results['artists']['items']) > 0)
+        self.assertTrue(results['artists']['items'][0]['name'] == 'Weezer')
+
```

Comment:
I added a test for this: `self.assertTrue(len(results_limited['artists']['items']) <= TOTAL)`. I have it checking for less than or equal to, since the result may not always find the total number requested.